enhancements/user-defined-default-ingress-controller.md

```diff
@@ -0,0 +1,262 @@
+   system to execute the Ingress Operator `render` command and move the output
+   into the correct location, causing the installer to apply the manifests
+   before the Ingress Operator's lifecycle begins.
+3. Add a installation procedure to the
+   [documentation](https://github.com/openshift/openshift-docs) describing how
+   to provide a default IngressController and reasons why one might want to do
+   so.
+
+### User Stories
+
+Generally speaking, the proposed capability generically improves support for
+many post-installation IngressController use cases. This section describes two
+possible use cases.
+
+#### Create an internal (non-public) cluster
+
+>As an OpenShift administrator, I want to create a new cluster which is never
+>exposed to the public internet.
+
+To accomplish this goal today on cloud platforms, users must (in addition to
+other non-ingress tasks):
+
+1. Install a cluster which unconditionally exposes ingress with a public load
+   balancer.
+2. Replace the default IngressController with an internal variant.
+
+This replacement procedure violates the constraints of the goal (as a public
+load balancer exists for some time). The procedure is also disruptive:
+
+1. During the replacement operation, default ingress is completely uninstalled
+   and then reinstalled, resulting in complete (but temporary) ingress
+   disruption.
+    * The old cloud load balancer is destroyed and a new load balancer is
+      provisioned.
+    * DNS records for the old cloud load balancer are destroyed and then
+      re-created for the new load balancer.
+
+If this proposal is implemented, during installation the user can provide the
+following IngressController manifest which satisfies the constraint as well as
+eliminating disruption and unnecessary (and possibly expensive) cloud resource
+provisioning.
+
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: IngressController
+metadata:
+  namespace: openshift-ingress-operator
+  name: default
+spec:
+  endpointPublishingStrategy:
+    type: LoadBalancerService
+    loadBalancer:
+      scope: Internal
+```
+
+#### Pre-configuring shards
+
+>As an OpenShift administrator, I want to the default IngressController to only
+>admits OpenShift's default Routes
+
+To accomplish this goal today, users must:
+
+1. Install a cluster which by default produces an IngressController which will
+   admit any Route resource in any namespace.
+2. Update the default IngressController to specify Route selector criteria which
+   exclude user defined Routes (where user defined Route is a Route labeled to
+   indicate the Route is user-defined).
+
+The update procedure has at least two notable drawbacks:
+
+1. Changing the IngressController will result in a rollout (which today can be
+   disruptive).
+2. The IngressController may already have admitted some Routes which would have
```

Comment:
how much of a concern is it? wouldn't the only actors in the cluster right-after install be openshift-operators.

Also what if the user disallowed openshift components to create these resources, which will cause the install to fail...

Reply:
I agree, the use case is probably too contrived. Other than pre-configuring replicas, maybe that leaves only the publishing type as a legit use case for now?

Reply:
> I agree, the use case is probably too contrived. Other than pre-configuring replicas, maybe that leaves only the publishing type as a legit use case for now?

Doesn't look overly contrived to me. Admitting routes that should be denied is a real problem.